cli/Main.d.ts

```diff
@@ -1,15 +1,10 @@
-interface Renderer {
-    name: string;
-    extension: string;
-    aceMode: string;
-}
-
 type SourceCode = string;
 type ErrorMessage = string;
 
 interface Main {
     renderers: Renderer[];
     main(config: Config): Either<ErrorMessage, SourceCode>;
+    mainWithOptions(options: {[name: string]: string}): ((config: Config) => Either<ErrorMessage, SourceCode>);
```

Comment:
`Config` already is a bag of options that's designed to make `Main` easier to call from `quicktype.io` and our IDE extension. Why are you adding additional options outside of `Config`? It's totally fine to change `Config` and therefore `Main.main` – when I said it would 'break' `quicktype.io`, I didn't mean you should not do it!